HPXMLtoOpenStudio/resources/battery.rb

```diff
@@ -37,6 +37,15 @@ def self.apply(runner, model, pv_systems, battery, schedules_file, unit_multipli
 
     return if rated_power_output <= 0 || nominal_capacity_kwh <= 0 || battery.nominal_voltage <= 0
 
+    if battery.is_shared_system
+      # Apportion to single dwelling unit by # bedrooms
+      fail if battery.number_of_bedrooms_served.to_f <= nbeds.to_f # EPvalidator.xml should prevent this
+
+      nominal_capacity_kwh = nominal_capacity_kwh * nbeds.to_f / battery.number_of_bedrooms_served.to_f
+      usable_capacity_kwh = usable_capacity_kwh * nbeds.to_f / battery.number_of_bedrooms_served.to_f
+      rated_power_output = rated_power_output * nbeds.to_f / battery.number_of_bedrooms_served.to_f
+    end
```

Comment:
I took the new base-bldgtype-mf-unit-shared-pv-battery.xml sample file, temporarily changed the shared battery so that all inputs (NominalCapacity, UsableCapacity, RatedPowerOutput, and NumberofBedroomsServed) were multiplied by 10, and ran it. Simulation results were identical to the original file.